### DISCUSSION TITLE
How to test retry behaviour?

### DISCUSSION THREAD

**Discussion by boredland** [quirrel-dev/quirrel#814] [Q&A]
Hi there! I'm tinkering around with squirrel and very happy so far. I wanted to try what happens when a job fails / test the retry behavior. I tried to throw inside the worker function, but this did not lead to a retry.

**Comment by Skn0tt** [comment]
Hi! Did you specify the [`retry` field](https://docs.quirrel.dev/api/queue#retryable)? (If you suspect a bug, please open an issue and provide a reproduction case so I can help you.)

**Reply by captain-http** [reply]
@Skn0tt I suspect there is a bug with the `retry` option. The documentation mentions that it's possible to pass the `retry` option as an argument to the `Queue` constructor.

> You can also specify `{ retry: [ ... ] }` as the third argument to the Queue constructor.

I think the `retry` option it's being ignored unless it's used in the `enqueue` function. I'll try later to isolate this bug and submit a POC in an issue.

**Reply by boredland** [reply]
I think I may have reproduced it here: https://github.com/quirrel-dev/quirrel/issues/815
